# Sweetie Bot base movement request.
# 
# * gait_type --- one of possible gaits.
# * duration --- movement duration
# * steps --- the number of step cycles.
# * base_goal --- goal pose of the robot base in header.frame_id frame.
# * base_goal_bounds --- defines which components of goal pose are fixed or fixed.
#      Zero bits indicate fixed coordinates, coordinates order (from low to high) is (R,Y,P,X,Y,Z).
# * ee_goal --- goal poses for end effector. For free gait their order defines step sequece.
# * position_tolerance --- admissable position error.
# * orientation_tolerance  --- admissable orientaton error.
#
Header header

string gait_type
float64 duration
uint32 n_steps

geometry_msgs/Pose base_goal
uint32 base_goal_bounds
sweetie_bot_clop_generator/EndEffectorGoal[] ee_goal

float64 position_tolerance
float64 orientation_tolerance

bool visualize_only

uint32 POSE_FIXED = 0
uint32 POSE_FREE_XY = 3
uint32 POSE_FREE_Z = 4
uint32 POSE_FIXED_POSITION = 56
uint32 POSE_FIXED_ORIENTATION = 7
uint32 POSE_FREE = 63
---
int32 error_code
string error_string

int32 SUCCESS = 0
int32 SOLUTION_NOT_FOUND = -1 #
int32 INVALID_GOAL = -2
int32 INTERNAL_ERROR = -3
int32 TOLERANCE_VIOLATED = -4
int32 INVALID_INITIAL_POSE = -5
int32 EXECUTION_FAILED = -6
---

float64 time_from_start
